### DISCUSSION TITLE
Help to display real time clock on VGA monitor.

### DISCUSSION THREAD

**Discussion by j3SuSs** [fdivitto/FabGL#254] [Q&A] [answered]
Hello, I am new to using this excellent library but I still have problems using it, I would like to be able to display a real-time clock on the entire screen, I have managed to increase the font size using external fonts, but when I want to display a counter using Canvas, the images overlap, not allowing me to visualize anything.
Thanks.

**Comment by fdivitto** [comment, marked as answer]
Please post source code

**Reply by j3SuSs** [reply]
#include "fabgl.h"
#define FABGL_FONT_INCLUDE_DEFINITION
#include "fonts/ariblk60.h"
#include "fonts/ariblk44.h"
#include "fonts/ariblk48.h"

fabgl::VGA16Controller DisplayController;
Canvas cv(&DisplayController);
int i=0;

void setup() {
  // put your setup code here, to run once:
  DisplayController.begin();
  DisplayController.setResolution(VGA_400x300_60Hz);
  //DisplayController.setResolution(VGA_320x200_70Hz); 
  //image for backgroud
  cv.setBrushColor(Color::Blue);
  cv.clear();

  cv.setPenColor(RGB888(64, 64, 0));
  cv.setPenWidth(5);

  cv.drawLine(0, 85, cv.getWidth(), 85);
  // cv.drawLine(0, 215, cv.getWidth(), 215);

  cv.setPenColor(RGB888(128, 0, 0));
  cv.drawRectangle(300, 160, 400, 240);

  cv.setPenColor(Color::White);
}

void loop() {
  // put your main code here, to run repeatedly:
  cv.setScrollingRegion(5, 5, 300, 80);
  cv.scroll(0,55);
  cv.selectFont(&fabgl::FONT_Arial48);
  cv.drawTextFmt(5, 10, "%02d:%02d", i,i);
  i++;
  delay(1000);
}

**Comment by fdivitto** [comment]
You have to re-fill the background for each iteration, otherwise you overlap graphics. setScrollingRegion() and scroll() are useful to scroll, not to fill background.
Try this (I've simplified a bit your code):
```
#include "fabgl.h"
#define FABGL_FONT_INCLUDE_DEFINITION
#include "Arial48.h"

fabgl::VGA16Controller DisplayController;
Canvas cv(&DisplayController);
int i=0;

void setup() {
  DisplayController.begin();
  DisplayController.setResolution(VGA_400x300_60Hz);

  cv.setBrushColor(Color::Blue);
  cv.clear();

  cv.setPenColor(Color::White);
  cv.selectFont(&fabgl::FONT_Arial48);
}

void loop() {
  cv.fillRectangle(5, 10, 130, 60);
  cv.drawTextFmt(5, 10, "%02d:%02d", i,i);
  i++;
  delay(1000);
}
```
[test.zip](https://github.com/fdivitto/FabGL/files/8980650/test.zip)

**Reply by j3SuSs** [reply]
Thanks for your reply, I'll try to do it that way.
